spec/features/posts_spec.rb

```diff
@@ -0,0 +1,104 @@
+  end
+  before { subject.save }
+
+  describe 'index page' do
+    it "I can see the user's username." do
+      visit user_posts_path(user.id)
+      page.has_content?(user.name)
+    end
+
+    it "I can see the user's profile picture." do
+      visit user_posts_path(user.id)
+      page.has_css?('.img-fluid')
+    end
+
+    it 'I can see the number of posts the user has written.' do
+      visit user_posts_path(user.id)
+      page.has_content?(user.posts_counter)
+    end
+
+    it "I can see a post's title." do
+      visit user_posts_path(user.id)
+      page.has_content?(subject.title)
+    end
+
+    it "I can see some of the post's body." do
+      visit user_posts_path(user.id)
+      page.has_content?(subject.text)
+    end
+
+    it 'I can see the first comments on a post.' do
+      comment = Comment.create(author_id: user.id, post_id: subject.id, text: 'Nice post')
+      visit user_posts_path(user.id)
+      page.has_content?(comment.text)
+    end
+
+    it 'I can see how many comments a post has.' do
+      visit user_posts_path(user.id)
+      page.has_content?(subject.comments_counter)
+    end
+
+    it 'I can see how many likes a post has.' do
+      visit user_posts_path(user.id)
+      page.has_content?(subject.likes_counter)
+    end
+
+    it 'I can see a section for pagination if there are more posts than fit on the view.' do
+      visit user_posts_path(user.id)
+      page.has_button?('Pagination')
+    end
+
+    it "When I click on a post, it redirects me to that post's show page." do
+      visit user_posts_path(user.id)
+      click_on subject.text
+      visit user_post_path(user.id, subject.id)
+      page.has_content?(subject.text)
+    end
+  end
+
+  describe 'show page' do
+    it "I can see a post's title." do
+      visit user_post_path(user.id, subject.id)
+      page.has_content?(subject.title)
+    end
+
+    it 'I can see how many comments a post has.' do
+      visit user_post_path(user.id, subject.id)
+      page.has_content?(subject.comments_counter)
+    end
+
+    it 'I can see how many likes a post has.' do
+      visit user_post_path(user.id, subject.id)
+      page.has_content?(subject.likes_counter)
+    end
+
+    it 'I can see who wrote the post.' do
+      visit user_post_path(user.id, subject.id)
+      page.has_content?(user.name)
+    end
+
+    it "I can see some of the post's body." do
+      visit user_post_path(user.id, subject.id)
+      page.has_content?(subject.text)
+    end
+
+    it 'I can see the username of each commentor.' do
+      comment = Comment.create(author_id: user.id, post_id: subject.id, text: 'Nice post')
+      visit user_post_path(user.id, subject.id)
+      page.has_content?(comment.author.name)
+    end
+    it 'I can see the comment each commentor left.' do
+      comment = Comment.create(author_id: user.id, post_id: subject.id, text: 'Nice post')
+      visit user_post_path(user.id, subject.id)
+      page.has_content?(comment.text)
+    end
+  end
+end
```

Comment:
Hi @megha-n-bodke 
You have done a great job implementing the different User post tests

- [ ] Please create a separate file for post index page and post show page

- Has featured tests for User post index page:

- Has featured tests for User post show page: